shacl12-ui/index.html

```diff
@@ -801,8 +801,592 @@ <h3>Document Conventions</h3>
+            constraint, or (at minimum) only allows classes as values.
+          </p>
+          <p>
+            <b>Rendering:</b>
+            This may be an auto-complete widget to select a class or a class hierarchy widget, or a combination thereof.
+            The permissible values are a given class or its subclasses, defaulting to <code>rdfs:Resource</code>.
+            This is typically used with <code>dash:rootClass</code> to allow the user to select a subclass of the given root class.
+          </p>
+          <pre class="example-shapes">
+ex:Drug-impactedCell
+	a sh:PropertyShape ;
+	sh:path ex:impactedCell ;
+	<b>dash:rootClass obo:CL_0000000 ;
+	shui:editor shui:SubClassEditor ;</b>
+	...</pre>
+        </section>
+        <section id="TextAreaEditor">
+          <h4>shui:TextAreaEditor</h4>
+          <p>
+            <b>Score:</b>
+            0 if the property is marked <code>sh:singleLine true</code>.
+            20 if the value is an <code>xsd:string</code> literal and <code>sh:singleLine false</code>.
+            5 if the value is an <code>xsd:string</code> literal.
+            2 if the property has <code>xsd:string</code> among the permissible datatypes.
+            <code>null</code> if the property has a custom datatype (not from xsd or rdf namespaces but for example <code>geo:wktLiteral</code>).
+            0 otherwise.
+          </p>
+          <p>
+            <b>Rendering:</b>
+            a multi-line text area to enter the value of a literal.
+          </p>
+          <img src="images/editors/TextAreaEditor.png" />
+          <pre class="example-shapes">
+ex:Country-description
+	a sh:PropertyShape ;
+	sh:path ex:description ;
+	<b>sh:datatype xsd:string ;
+	sh:singleLine false ;</b>
+	...</pre>
+        </section>
+        <section id="TextAreaWithLangEditor">
+          <h4>shui:TextAreaWithLangEditor</h4>
+          <p>
+            <b>Score:</b>
+            0 if the property is marked <code>sh:singleLine true</code>.
+            15 if the value is an <code>rdf:langString</code> literal and <code>sh:singleLine false</code>.
+            5 if the value is an <code>rdf:langString</code> literal or the property permits such values.
+            0 otherwise.
+          </p>
+          <p>
+            <b>Rendering:</b>
+            a multi-line text area to enter the value of a literal and a drop-down to select a language.
+          </p>
+          <img src="images/editors/TextAreaWithLangEditor.png" />
+          <pre class="example-shapes">
+ex:Country-description
+	a sh:PropertyShape ;
+	sh:path ex:description ;
+	<b>sh:datatype rdf:langString ;
+	sh:singleLine false ;</b>
+	...</pre>
+        </section>
+        <section id="TextFieldEditor">
+          <h4>shui:TextFieldEditor</h4>
+          <p>
+            <b>Score:</b>
+            10 if the value is a literal that is neither <code>rdf:langString</code> nor <code>xsd:boolean</code>.
+            0 otherwise.
```

Comment:
Another global suggestion. Make the score values `<code>` to set them slightly apart from the running text.
```suggestion
            <code>10</code> if the value is a literal that is neither <code>rdf:langString</code> nor <code>xsd:boolean</code>.
            <code>0</code> otherwise.
```